pylint/checkers/classes/class_checker.py

```diff
@@ -2034,11 +2037,19 @@ def _is_mandatory_method_param(self, node):
 
         Name is `self` for method, `cls` for classmethod and `mcs` for metaclass.
         """
-        return (
-            self._first_attrs
-            and isinstance(node, nodes.Name)
-            and node.name == self._first_attrs[-1]
-        )
+        if self._first_attrs:
```

Comment:
Could you add typing to this function signature? You probably know what this accepts and does after changing it a bit :)
Should be:
`_is_mandatory_method_param(self, node: nodes.NodeNG) -> bool`

While we're at it, we can also add typing to `_is_type_self_call`?
Should be:
`def _is_type_self_call(self, expr: nodes.NodeNG) -> bool`
If `mypy` starts to complain about `attribute.expr` possible being `None` on L973 we can add a `type: ignore` for it. That should be fixed by `astroid` typing.